docs/2_architecture/agg_mode_components/1_deep_dive.md

```diff
@@ -0,0 +1,131 @@
+
+## Aggregators and Supported Proof Types
+
+Two separate aggregators are run every 24 hours:
+
+-   **Risc0**: Aggregates proofs of types `Composite` and `Succinct`.
+-   **SP1**: Aggregates proofs of type `Compressed`.
+
+## Proof Commitment
+
+The **proof commitment** is a hash that uniquely identifies a proof. It is defined as the keccak of the proof public inputs + program ID:
+
+-   **For SP1**:  
+    The commitment is computed as: `keccak(proof_public_inputs_bytes || vk_hash_bytes)`
+-   **For Risc0**:  
+    The commitment is computed as: `keccack(receipt_public_inputs_bytes || image_id_bytes)`
+
+## Multilayer Aggregation
+
+To scale aggregation without exhausting zkVM memory, aggregation is split in two programs:
+
+1. **User Proof Aggregator**  
+   Processes chunks of `n` user proofs. Each run creates an aggregated proof that commits to a Merkle root of the user proofs inputs. This step is repeated for as many chunks as needed. Usually each chunks contains `256` proofs but it can be lowered based on the machine specs.
+
+2. **Chunk Aggregator**  
+   Aggregates all chunk-level proofs into a single final proof. It receives:
+
+    - The chunked proofs
+    - The original [proofs commitments](#proof-commitment) included each chunk received
+
+    During verification, it checks that each chunk’s committed Merkle root matches the reconstructed root to ensure input correctness. The final Merkle root, representing all user [proofs commitments](#proof-commitment), is then committed as a public input.
+
+## Verification
+
+Once aggregated, the proof is sent to Ethereum and verified via the `AlignedProofAggregationService` contract. Depending on the proving system, the contract invokes:
+
+-   `verifySP1` for SP1 proofs
+-   `verifyRisc0` for Risc0 proofs
+
+Each function receives:
+
+-   The public inputs
+-   The proof binary
+
+The program ID is hardcoded in the contract to ensure only trusted aggregation programs (`chunk_aggregator`) are accepted.
+
+If verification succeeds, the new proof is added to the `aggregatedProofs` map in contract storage.
+
+### Proof Inclusion Verification
+
+To verify a user’s proof on-chain, the following must be provided:
+
+-   The proof bytes
+-   The proof public inputs
+-   The program ID
+-   A Merkle proof
+
+The Merkle root is computed and checked for existence in the contract using the `verifyProofInclusion` function of the `ProofAggregationServiceContract`, which:
+
+1. Computes the merkle root
+2. Returns `true` or `false` depending if there exists an `aggregatedProof` with the computed root.
+
+## Data Availability
+
+When submitting the aggregated proof to Ethereum, we include a **blob** that contains the [commitments](#proof-commitment) of all the individual proofs that were aggregated. This blob serves two main purposes:
```

Comment:
Everything from this can be removed, it's not information for curious users, at this point it's just implementation details